### PR TITLE
support Scala 2.12.16 (and 2.12.17 nightlies)

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

File: src/test/resources/patches/2.12.17.patch [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
++++ target/expected_output/basic/akka/persistence/testkit/EventStorage.java	2020-03-23 16:04:03.313049968 +0100
+@@ -5,7 +5,9 @@
+       * Static reference to the singleton instance of this Scala object.
+       */
+     public static final JournalPolicies$ MODULE$ = null;
++    public  akka.persistence.testkit.ProcessingPolicy.DefaultPolicies<akka.persistence.testkit.JournalOperation>.PassAll$ PassAll ()  { throw new RuntimeException(); }
+     public   JournalPolicies$ ()  { throw new RuntimeException(); }
+   }
+-  public  akka.persistence.testkit.ProcessingPolicy.DefaultPolicies<akka.persistence.testkit.JournalOperation>.PassAll$ DefaultPolicy ()  ;
++  public  void akka$persistence$testkit$EventStorage$_setter_$DefaultPolicy_$eq (akka.persistence.testkit.ProcessingPolicy.DefaultPolicies<akka.persistence.testkit.JournalOperation>.PassAll x$1)  ;
```
Comment (raboof) [inline src/test/resources/patches/2.12.17.patch:69]: shouldn't this have been marked `ACC_SYNTHETIC`? I guess not important for this PR but a potential upstream problem

File: src/test/resources/patches/2.12.17.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
++    public  void akka$actor$dsl$Inbox$InboxExtension$_setter_$DSLInboxQueueSize_$eq (int x$1)  ;
+   }
+ }
+--- target/expected_output/basic/akka/persistence/testkit/EventStorage.java	2020-03-23 16:04:03.313049968 +0100
++++ target/expected_output/basic/akka/persistence/testkit/EventStorage.java	2020-03-23 16:04:03.313049968 +0100
+@@ -5,7 +5,9 @@
+       * Static reference to the singleton instance of this Scala object.
+       */
+     public static final JournalPolicies$ MODULE$ = null;
++    public  akka.persistence.testkit.ProcessingPolicy.DefaultPolicies<akka.persistence.testkit.JournalOperation>.PassAll$ PassAll ()  { throw new RuntimeException(); }
```
Comment (raboof) [inline src/test/resources/patches/2.12.17.patch:65]: This method doesn't seem to show up in `javap`, so it looks like a bug in genjavadoc that this now shows up?

**Comment by raboof** [comment]
I guess we can merge this anyway, since we're not releasing for 2.12.17 yet.

**Comment by SethTisue** [comment]
@raboof I'll investigate issues you raise (good eye!), but they aren't new in 2.12.16; I just copied the patch file from the 2.12.15 release. (I'll need to do some digging to figure out which 2.12.x version(s) were the actual starting point.)

given that, I suggest you go ahead and release anyway, and if my investigation determines that we need to make changes to genjavadoc, we'll do that later.

**Comment by SethTisue** [comment]
so it turns out that the two questionable methods are there even in `src/test/resources/patches/2.12.3.patch`, so whatever the problem is, it's quite ancient

given that, I'm not motivated to investigate further. if it had been something that changed in Scala in the last 1–2 years, say, I might have been interested because I'd want to see if it was a Scala regression. but this looks more like a quality problem with genjavadoc. unless it's showing up in the Akka docs and confusing actual users, it doesn't seem worth digging into further

**Pull request merged**